web_src/js/modules/observer.ts

```diff
@@ -1,52 +1,71 @@
 import {isDocumentFragmentOrElementNode} from '../utils/dom.ts';
 
-type DirElement = HTMLInputElement | HTMLTextAreaElement;
+let globalSelectorObserverInited = false;
 
-// for performance considerations, it only uses performant syntax
-function attachDirAuto(el: Partial<DirElement>) {
-  if (el.type !== 'hidden' &&
-      el.type !== 'checkbox' &&
-      el.type !== 'radio' &&
-      el.type !== 'range' &&
-      el.type !== 'color') {
-    el.dir = 'auto';
-  }
-}
+type SelectorHandler = {selector: string, handler: (el: HTMLElement) => void};
+const selectorHandlers: SelectorHandler[] = [];
+
+type GlobalEventFunc<T extends HTMLElement, E extends Event> = (el: T, e: E) => (void | Promise<void>);
```

Comment:
```suggestion
type GlobalEventFunc<T extends HTMLElement, E extends Event> = (el: T, e: E) => (Promisable<void>);
```

Import `Promisable` from `type-fest`.